src/tests/test_semantics.py

```diff
@@ -1,18 +1,67 @@
 import itertools
 import pandas as pd
+import pytest
 
-from altk.language.semantics import Universe
-
+from ultk.language.semantics import Universe
+from ultk.language.semantics import Meaning
+from ultk.language.semantics import Referent
 
 forces = ("weak", "strong")
 flavors = ("epistemic", "deontic")
 
 
-def test_universe_from_df():
+class TestSemantics:
     points = [
         {"name": f"{force}+{flavor}", "force": force, "flavor": flavor}
         for (force, flavor) in itertools.product(forces, flavors)
     ]
     dataframe = pd.DataFrame(points)
     universe = Universe.from_dataframe(dataframe)
-    assert points == [referent.__dict__ for referent in universe.referents]
+
+    ref1 = Referent(
+        name="weak+epistemic", properties={"force": "weak", "flavor": "epistemic"}
+    )
+
+    def test_universe_from_df(self):
+        assert TestSemantics.points == [
+            referent.__dict__ for referent in TestSemantics.universe.referents
+        ]
+
+    def test_referent_match(self):
+        ref2 = Referent(
+            name="weak+epistemic", properties={"force": "weak", "flavor": "epistemic"}
+        )
+        assert TestSemantics.ref1 == ref2
+
+    def test_referent_mismatch(self):
+        ref3 = Referent(
+            name="neutral+epistemic",
+            properties={"force": "neutral", "flavor": "epistemic"},
+        )
+        assert TestSemantics.ref1 != ref3
+
+    def test_universe_match(self):
+        second_dataframe = pd.DataFrame(TestSemantics.points)
```

Comment:
Might be nice to make `second_dataframe` using a different initializer than the `from_dataframe`, i.e. just using the list of referents